src/psyclone/gocean1p0.py

```diff
@@ -764,13 +771,10 @@ def _lower_bound(self):
     # -------------------------------------------------------------------------
     def __str__(self):
         ''' Returns a string describing this Loop object '''
-        step = self._step
-        if not step:
-            step = "1"
+        step = str(self.step_expr)
 
-        result = ("Loop[" + self._id + "]: " + self._variable_name +
-                  "=" + self._id + " lower=" + self._lower_bound() +
-                  "," + self._upper_bound() + "," + step + "\n")
+        result = "Loop[id='" + self._id + "', "
+        result += "variable='" + self._variable_name + "']\n"
```

Comment:
The idea is that they are already shown in the children below :
```
Loop [...]
    Literal['1']  <- Lower_bound
    Literal['N']  <- Upper bound
    ...
```
However, in the code you reviewed this could have been 'Uninitialised'. I have now make sure that the `self._lower/upper_bound` are called at the beginning of the `__str__`, even if we call them redundantly (but this should be fixed by #440)

Also I am calling now the parent `__str__`